source/louisHelper.py

```diff
@@ -23,6 +31,77 @@
 	louis.LOG_FATAL: log.ERROR,
 }
 
+
+# Note: liblouis table resolvers return char**,
+# but POINTER(c_char_p) is unsupported as a ctypes callback return type.
+# C901 '_resolveTable' is too complex due to several statements for debugging purposes.
+@WINFUNCTYPE(c_void_p, c_char_p, c_char_p)
+def _resolveTable(tablesList: bytes, base: bytes | None) -> int | None:  # noqa: C901
+	"""Resolve braille table file names to file paths.
+
+	Unlike the default table resolver from liblouis, this implementation does
+	not confer any special role to the directory of the first table of the list
+	and completely ignores the the liblouis data path and the
+	C{LOUIS_TABLEPATH} environment variable.
+	Instead, when base is None, it fetches the tables as registered in the brailleTables module,
+	If they point to an existing file, the value of the absolutePath property is returned.
+	When base is not None, the imported table is either looked up into the same directory as the base table,
+	or in the directory with the built-in tables.
+	"""
+	if _isDebug():
+		log.debug(f"liblouis called table resolver wit params: tablesList={tablesList}, base={base}")
+	tables = tablesList.decode(louis.fileSystemEncoding).split(",")
+	if base is not None:
+		base: str = base.decode(louis.fileSystemEncoding)
+	paths = []
+	for table in tables:
+		if _isDebug():
+			log.debug(f"Resolving {table!r}")
+		resolved = False
+		if base is None:
+			try:
+				registeredTable = brailleTables.getTable(table)
+				path = registeredTable.absolutePath
+			except LookupError:
+				if _isDebug():
+					log.debug(f"Table {table!r} not registered, falling back to built-in table lookup")
+				path = os.path.join(brailleTables.TABLES_DIR, table)
+			if os.path.isfile(path):
+				paths.append(path.encode(louis.fileSystemEncoding))
+				if _isDebug():
+					log.debug(f"Resolved {table!r} to {path!r}")
+				resolved = True
+		else:
+			directoriesToSearch = [os.path.dirname(base)]
+			if brailleTables.TABLES_DIR not in directoriesToSearch:
+				directoriesToSearch.append(brailleTables.TABLES_DIR)
+			for directory in directoriesToSearch:
+				path = os.path.join(directory, table)
+				if os.path.isfile(path):
+					paths.append(path.encode(louis.fileSystemEncoding))
+					if _isDebug():
+						log.debug(f"Resolved {table!r} to {path!r} for base {base!r}")
+					resolved = True
+					break
+		if not resolved:
+			if _isDebug():
+				log.error(f"Could not resolve table {table!r}")
```

Comment:
If it is an error, I'm thinking it should always be logged, not only when debug is on.